### DISCUSSION TITLE
How to run shairport with dbus support?

### DISCUSSION THREAD

**Discussion by killerlife** [mikebrady/shairport-sync#1862] [Q&A]
I was configure with --with-dbus-interface and --with-dbus-test-client.
It print a  warning: could not acquire a Shairport Sync native D-Bus interface "org.gnome.ShairportSync.i73906" on the system bus after I run shairport-sync.

I get nothing while I run the shairport-sync-dbus-test-client. 

Maybe I need to write a dbus configure file. Is anyone tell me how to do this?

Thanks.

**Comment by mikebrady** [comment]
Thanks for the post, but this is more of an issue than a discussion topic. If you want to continue the conversation, please close this and open an issue.

The `# make install` step takes care of all this and installs a D-Bus configuration file. Service provision on the D-Bus system bus is tightly controlled. With the installed service file, Shairport Sync must run under the user `shairport-sync` or as `root` (not recommended) to be allowed provide the D-Bus system service. 
After installation, you might need to reboot the system for it to recognise the the new D-Bus service file.

The warning:
```
could not acquire a Shairport Sync native D-Bus interface "org.gnome.ShairportSync.i73906"
```
Is indicating that Shairport Sync doesn't have permission  to provide a service on the system bus -- is it running from the command line?

Alternatively you can get Shairport Sync to provide a service on the D-Bus session bus -- no security issues. Details are in the Shairport Sync configuration file.

**Reply by killerlife** [reply]
Hi Mark, thank you for your advise. It's my mistake. It's work after I add my user policy into shairport-sync-dbus.conf. Because I run shairport with another user, not root and shairport-sync.
